### PR TITLE
export Stanza based receiver metrics

### PR DESCRIPTION
**Pull request by emaderer**
Added receiver metrics (accepted, refused, etc) to receivers that are based on Stanza

### CODE REVIEW

File: internal/stanza/factory.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ func createLogsReceiver(logReceiverType LogReceiverType) receiverhelper.CreateLo
 		nextConsumer consumer.Logs,
 	) (component.LogsReceiver, error) {
 		inputCfg, err := logReceiverType.DecodeInputConfig(cfg)
+
```
Comment (dmitryax) [inline internal/stanza/factory.go:54]: Nit: please remove this added line

File: CHANGELOG.md [hunk 2]
```diff
@@ -61,9 +62,9 @@
 - Use go mod compat, drops support for reproducibility with go 1.16 (#7915)
 - `apachereceiver`: Update instrumentation library name from `otel/apache` to `otelcol/apache` (#7754)
 - `pkg/translator/prometheusremotewrite`: Cleanup prw translator public functions (#7776)
-- `prometheusreceiver`: The OpenCensus-based metric conversion pipeline has 
+- `prometheusreceiver`: The OpenCensus-based metric conversion pipeline has
```
Comment (dmitryax) [inline CHANGELOG.md:65]: Can you remove or submit another PR for the unrelated formatting fixes?

**Review by dmitryax — COMMENTED** [review]
LGTM. just couple nits

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@emaderer there are some linter issues blocking the build

File: internal/stanza/receiver.go [hunk 3]
```diff
@@ -113,21 +115,23 @@ func (r *receiver) consumerLoop(ctx context.Context) {
...
 		select {
 		case <-doneChan:
 			r.logger.Debug("Consumer loop stopped")
 			return
 
 		case pLogs, ok := <-pLogsChan:
+			obsrecvCtx := r.obsrecv.StartLogsOp(ctx)
 			if !ok {
 				r.logger.Debug("Converter channel got closed")
 				continue
```
Comment (pmm-sumo) [inline internal/stanza/receiver.go:128]: Should we always expect that EndLogsOp is called for each StartLogsOp? In this path this would not be the case and I am not sure if that's ok or not. I looked at EndsLogsOp and I think that the only meaningful result of that would be not ending the observability span explicitly, so not sure if it hurts (that would be rather minor) or not but thought it might be worth calling out

Comment (dmitryax) [reply internal/stanza/receiver.go:128]: Good point. I agree that `obsrecvCtx := r.obsrecv.StartLogsOp(ctx)` should be placed right before `cErr := r.consumer.ConsumeLogs(ctx, pLogs)`

Comment (emaderer) [reply internal/stanza/receiver.go:128]: Thanks!

**Review by pmm-sumo — APPROVED** [review]
Seems some minor conflicts need to be resolved

**Review by djaglowski — APPROVED** [review]
